### PR TITLE
Remove rspec-doc-template yard plugin + remove empty lines from HTML doc

### PR DESCRIPTION
**Pull request by benoittgt**
* Remove rspec-doc-template yard plugin. Impact: https://github.com/rspec/rspec.github.io/pull/138
* Remove empty lines generated by rdoc/erb. Impact: https://github.com/rspec/rspec.github.io/pull/139 ⚠️

Related:
- https://github.com/rspec/rspec-rails/pull/2298#discussion_r400436204

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ def run_command(command, opts={})
     FileUtils.cd(path) do
       puts "#{'='*3} #{dir} #{'='*(40 - dir.length)}"
       begin
-        Bundler.clean_system(command)
+        Bundler.unbundled_system(command)
```
Comment (pirj) [inline Rakefile:29]: 👍

Comment (JonRowe) [reply Rakefile:29]: Hm theres a reason we were using clean_system but I've forgotten what it is

Comment (benoittgt) [reply Rakefile:29]: In my case, I just followed the deprecation warning.

Comment (JonRowe) [reply Rakefile:29]: Ah then I think its a bundler 1 vs 2 issue

**Comment by benoittgt** [comment]
Sorry, last night I remember that my script to remove lines is not scoped to the rspec gem version you want in `source/` folder in rspec.github.io. There is the same behavior for the two other bash commands.

I added [a commit](https://github.com/rspec/rspec-dev/pull/245/commits/40208605a125623595bf4a3bc51cf18324598e0d) that do `sed` changes only on the gems and version that you asked. This will avoid extra work on all documentation every time we publish.

This was maybe a choice that is wanted. I can revert it. 😊

I will keep the all lines already removed in rspec.gihub.io because it is a necessary cleaning.

The last commit do not change the doc generated (expect doc generation timestamp in footer).

File: Rakefile (hunk 1 above)
Comment (benoittgt) [reply Rakefile:29]: Is this a blocker for this PR? 

On another project we added an helper method that check if `unbundled_system` was defined, if not, fallback to `clean_system`. But I think we can keep it like this.

I didn't try all the rake commands and I lack history on the subject so feel free to tell me Jon.

Comment (JonRowe) [reply Rakefile:29]: No it's our dev tools, we can always revert.

**Comment by benoittgt** [comment]
Thanks Jon :)

**Pull request merged**